proto/src/domain/notes.rs

```diff
@@ -28,6 +28,7 @@ impl From<(usize, usize, NoteEnvelope)> for note::NoteCreated {
             note_id: Some(note.note_id().into()),
             sender: Some(note.metadata().sender().into()),
             tag: note.metadata().tag().into(),
+            details: None,
```

Comment:
When do we use this conversion? It may be good to add a comment on why it is ok to always set `details` to `None`.

Reply:
Also consider adding `Option<Vec<u8>>` as `details` argument for conversion:
```rust
impl From<(usize, usize, NoteEnvelope, Option<Vec<u8>>)> for note::NoteCreated {
    fn from((batch_idx, note_idx, note, details): (usize, usize, NoteEnvelope, Option<Vec<u8>>)) -> Self {
```
It would be more flexible and clearer to understand.

Reply:
// This is `None` for now as this conversion is used by the block-producer
            // when using `apply_block`. The block-producer has not yet been updated to support
            // on-chain notes. Will be set to the correct value when the block-producer is
            // up-to-date.